### PR TITLE
Bolg_App_Rails

### PR DESCRIPTION
**Pull request by AhmadiGu**
### N+1_problem_spec branch
*The following changes were introduced:* 
**Important** kindly check this commit I repeat the week and this branch approved
https://github.com/AhmadiGu/Bolg_App_Rails/pull/9/commits/4aba0aa57d07212c8ede43a97790b76fdd274f9c

- Use include for comments in the post controller  to solve n+1
-  Write the test for  the Post view   
      - index
      - show
-  Write the test for  the User view   
      - index
      - show
 
Before 
Screenshot (433)

After 
Screenshot (434)

### CODE REVIEW

File: spec/views/user/index.html.erb_spec.rb [hunk 1]
```diff
@@ -4,11 +4,11 @@
...
+                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', email: 'example1@example', posts_counter: 0)
       @user2 = User.create(name: 'Ahmad',
-                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', posts_counter: 0)
+                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', email: 'example2@example', posts_counter: 0)
       @user3 = User.create(name: 'Muzafar',
-                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', posts_counter: 0)
+                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', email: 'example3@example', posts_counter: 0)
 
       visit users_path
     end
```
Comment (zieeco) [inline spec/views/user/index.html.erb_spec.rb]: ### USE CAPYBARA TO WRITE INTEGRATION TESTS FOR EACH VIEW IN YOUR PROJECT

- [ ] Hey, you're meant to use capybara to write tests for the following `User index page`, `user show page`, 
`Post show page`, `User post index page`. Currently, you've not added any of these tests. kindly ensure you've added them. :pray: The test file should look something like `spec/features/post_index_spec.rb`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,






Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user/index.html.erb_spec.rb (hunk 1 above)
Comment (AhmadiGu) [reply spec/views/user/index.html.erb_spec.rb]: Hi, hope to be fine all of these specs are in view/user and view/post files

Comment (BenMukebo) [reply spec/views/user/index.html.erb_spec.rb]: - [ ]  Great job implementing the integration test, but some of them still not passing, current m running 10 tests that fail,, please make sure to fix all failing tests 💯 👍🏻 Here is the screenshot 👇🏻



image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @AhmadiGu, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- Gitflow is followed. :heavy_check_mark:
-  Your App solves N+1 problem when fetching all posts and their comments . ✔️
- Has Capybara gem in `Gemfile` for integration test. ✔️  
- Your Pull Request has a proper title and description. ✔️  


 ## Required Changes ♻️
_Check the comments under the review._


You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user/index.html.erb_spec.rb [hunk 2]
```diff
@@ -4,11 +4,11 @@
...
+                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', email: 'example1@example', posts_counter: 0)
       @user2 = User.create(name: 'Ahmad',
-                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', posts_counter: 0)
+                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', email: 'example2@example', posts_counter: 0)
       @user3 = User.create(name: 'Muzafar',
-                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', posts_counter: 0)
+                           photo: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJHYna2rR4LtBNlA3gUSGxsmaiYY4PyxQEcA&usqp=CAU', bio: 'Spring Sun', email: 'example3@example', posts_counter: 0)
```
Comment (TedLivist) [inline spec/views/user/index.html.erb_spec.rb]: - [ ] The test in this file is failing.

  ffis

  If you look at your User model in the console there is no `email` attribute for the user but you are trying to create a user with this attribute.

  mduw

  You can fix this by removing `email` attributes from these lines.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @AhmadiGu and @shella12,

✔️ Integration tests

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @AhmadiGu,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linters passed ✔️ 
- Test passed ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**